src/aggregator/aggregator/entry.go

```diff
@@ -1076,8 +1076,14 @@ type aggregationValue struct {
 type aggregationValues []aggregationValue
 
 func (vals aggregationValues) index(k aggregationKey) int {
-	for i, val := range vals {
-		if val.key.Equal(k) {
+	// keep in sync with aggregationKey.Equal()
+	// this is >2x slower if not inlined manually.
+	for i := range vals {
```

Comment:
Hmm dunno if this is possible, but any way to add a fuzz test comparing behavior using equals vs using the manual inline here? Less to test the current version of the code than to ensure against future regressions haha.

Reply:
Also, maybe worth adding a pointer from `Equal` to here to make sure anyone changing that method also changes this one?

Makes one wish for macros...

Reply:
good point as I was just lazy, but then again I missed something in tests :D